From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 27 Feb 2020 13:34:08 +0000
Subject: ARM: dts: bcm283x: Restore simple-bus to /clocks

Commit [1] ([2] upstream) removed the "simple-bus" compatible string,
"#address-cells" and "#size-cells" to fix some DT scheme warnings. In
doing so it prevented the child clocks from being instantiated - the
node might as well be absent.

Fix the problem (at least with regards to functionality) by restoring
"simple-bus" and the sizes.

See: https://github.com/raspberrypi/linux/issues/3481
Fixes: [1] 238506ebdea7 ("ARM: dts: bcm283x: Remove simple-bus from fixed clocks")
Fixes: [2] 4b2d24662126 ("ARM: dts: bcm283x: Remove simple-bus from fixed clocks")

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm283x.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/bcm283x.dtsi b/arch/arm/boot/dts/bcm283x.dtsi
index 9cdae56704f5..3ad3e9ad5692 100644
--- a/arch/arm/boot/dts/bcm283x.dtsi
+++ b/arch/arm/boot/dts/bcm283x.dtsi
@@ -480,6 +480,10 @@
 	};
 
 	clocks {
+		compatible = "simple-bus";
+		#address-cells = <0>;
+		#size-cells = <0>;
+
 		/* The oscillator is the root of the clock tree. */
 		clk_osc: clk-osc {
 			compatible = "fixed-clock";
